From d25084c791d82850fb9ea0f3a5ad178e28991e4b Mon Sep 17 00:00:00 2001
From: Rohan Garg <rohan@garg.io>
Date: Mon, 13 Feb 2017 13:39:51 +0100
Subject: [PATCH 1/2] Figure out the GTK version instead of using a hardcoded
 one

Summary: Find the GTK3 pkg-config file and pick up a version using that.

Test Plan:
* Do a clean rebuild
* Picks up the right GTK3 version and installs the right CSS file.

Subscribers: plasma-devel

Tags: #plasma

Differential Revision: https://phabricator.kde.org/D4596
---
 CMakeLists.txt | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4dc23c5..db52ea5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,10 +7,14 @@ cmake_minimum_required(VERSION 2.8.12)
 find_package(ECM 0.0.9 REQUIRED NO_MODULE)
 include(FeatureSummary)
 
-set(WITH_GTK3_VERSION 3.18 CACHE STRING
-    "Use the theme for the specified GTK+3 version (default: 3.18)")
-
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR} "${CMAKE_SOURCE_DIR}/cmake")
+find_package(GTKEngine)
+
+find_package(PkgConfig)
+pkg_check_modules(PKG_GTK REQUIRED gtk+-3.0)
+
+set(WITH_GTK3_VERSION ${PKG_GTK_VERSION})
+
 include(KDEInstallDirs)
 
 # Exclude these patterns when installing directories as they have no use
@@ -25,6 +29,4 @@ add_subdirectory(Breeze-dark-gtk)
 add_subdirectory(Breeze-gtk)
 add_subdirectory(kconf_update)
 
-find_package(GTKEngine)
-
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
-- 
2.11.0

